ambry-quota/src/main/java/com/github/ambry/quota/capacityunit/JsonCUQuotaSource.java

```diff
@@ -0,0 +1,311 @@
+  }
+
+  @Override
+  public void charge(RestRequest restRequest, BlobInfo blobInfo, Map<QuotaName, Double> requestCostMap) {
+    String resourceId = QuotaUtils.getQuotaResourceId(restRequest);
+    if (!cuQuota.containsKey(resourceId)) {
+      return;
+    }
+    if (QuotaUtils.isReadRequest(restRequest) && requestCostMap.containsKey(QuotaName.READ_CAPACITY_UNIT)) {
+      cuUsage.get(resourceId).rcu += requestCostMap.get(QuotaName.READ_CAPACITY_UNIT);
+      feUsage.rcu += requestCostMap.get(QuotaName.READ_CAPACITY_UNIT);
+    } else if (requestCostMap.containsKey(QuotaName.WRITE_CAPACITY_UNIT)) {
+      cuUsage.get(resourceId).wcu += requestCostMap.get(QuotaName.WRITE_CAPACITY_UNIT);
+      feUsage.wcu += requestCostMap.get(QuotaName.WRITE_CAPACITY_UNIT);
+    }
+  }
+
+  @Override
+  public QuotaRecommendation checkResourceUsage(RestRequest restRequest) {
+    String resourceId = QuotaUtils.getQuotaResourceId(restRequest);
+    if (!cuQuota.containsKey(resourceId)) {
+      return null;
+    }
+    QuotaName quotaName = QuotaUtils.getQuotaName(restRequest);
+    long limit = cuQuota.get(resourceId).getQuotaValue(quotaName);
+    long usage = cuUsage.get(resourceId).getQuotaValue(quotaName);
+    return buildQuotaRecommendation(limit, usage, quotaName);
+  }
+
+  @Override
+  public QuotaRecommendation checkFrontendUsage(RestRequest restRequest) {
+    QuotaName quotaName = QuotaUtils.getQuotaName(restRequest);
+    return buildQuotaRecommendation(feQuota.getQuotaValue(quotaName), feUsage.getQuotaValue(quotaName), quotaName);
+  }
+
+  public void updateNewQuota(QuotaResource quotaResource, long rcu, long wcu) {
+    cuQuota.put(quotaResource.getResourceId(), new CUQuota(rcu, wcu));
+    cuUsage.put(quotaResource.getResourceId(), new CUQuota(0, 0));
+  }
+
+  public Map<String, CUQuota> getAllQuota() {
+    return cuQuota;
+  }
+
+  public Map<String, CUQuota> getAllQuotaUsage() {
+    return cuUsage;
+  }
+
+  private QuotaRecommendation buildQuotaRecommendation(long limit, long usage, QuotaName quotaName) {
+    boolean shouldThrottle = (usage >= limit);
+    return new QuotaRecommendation(shouldThrottle, (usage * 100) / (float) limit, quotaName,
+        shouldThrottle ? THROTTLE_HTTP_STATUS : NO_THROTTLE_HTTP_STATUS,
+        shouldThrottle ? THROTTLE_RETRY_AFTER_MS : NO_THROTTLE_RETRY_AFTER_MS);
+  }
+
+  public static class CUQuota {
+    static final String RCU_FIELD_NAME = "rcu";
+    static final String WCU_FIELD_NAME = "wcu";
+
+    private long rcu;
+    private long wcu;
+
+    public CUQuota() {
+    }
+
+    @JsonIgnore
+    public CUQuota(JsonNode jsonNode) {
+      this.rcu = jsonNode.get(RCU_FIELD_NAME).asLong();
+      this.wcu = jsonNode.get(WCU_FIELD_NAME).asLong();
+    }
+
+    @JsonIgnore
+    public CUQuota(long rcu, long wcu) {
+      this.rcu = rcu;
+      this.wcu = wcu;
+    }
+
+    @JsonIgnore
+    public static boolean isQuotaNode(JsonNode jsonNode) {
+      return jsonNode.has(WCU_FIELD_NAME);
+    }
```

Comment:
The constructor and the static method should be JsonIgnored automatically, right?